File: omnes.Web/Modules/ServerTypes/Parametros/TiposDocumentoForm.ts
import { StringEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface TiposDocumentoForm {
    NombreTipoDocumento: StringEditor;
    Baja: BooleanEditor;
    Sigla: StringEditor;
}

export class TiposDocumentoForm extends PrefixedContext {
    static formKey = 'Parametros.TiposDocumento';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TiposDocumentoForm.init)  {
            TiposDocumentoForm.init = true;

            var w0 = StringEditor;
            var w1 = BooleanEditor;

            initFormType(TiposDocumentoForm, [
                'NombreTipoDocumento', w0,
                'Baja', w1,
                'Sigla', w0
            ]);
        }
    }
}

File: omnes.Web/Modules/Parametros/TiposDocumento/TiposDocumentoForm.cs
using Serenity.ComponentModel;

namespace omnes.Parametros.Forms;

[FormScript("Parametros.TiposDocumento")]
[BasedOnRow(typeof(TiposDocumentoRow), CheckNames = true)]
public class TiposDocumentoForm
{
    public string NombreTipoDocumento { get; set; }
    public bool Baja { get; set; }
    public string Sigla { get; set; }
}